### PR TITLE
Fixing Racy ClusterLogVerboseEnabledSpec

### PR DESCRIPTION
**Pull request by eaba**
## Changes

* [Exp.] Increase `TimeOut` to 1 mins from default 3s
* Converted test to `async/await`
* Added `JoinAsync` and `DownAsync`

## Checklist

For significant changes, please ensure that the following have been completed (delete if not relevant):

* [x] I have [reviewed my own pull request](https://getakka.net/community/contributing/index.html#review-your-own-pull-requests).

### CODE REVIEW

File: src/core/Akka.Cluster.Tests/ClusterLogSpec.cs [hunk 1]
```diff
@@ -58,13 +59,13 @@ protected void AwaitUp()
...
+        protected async Task Join(string expected)
         {
-            Within(TimeSpan.FromSeconds(10), () =>
+            await WithinAsync(TimeSpan.FromSeconds(10), async () =>
             {
-                EventFilter
+                await EventFilter
                     .Info(contains: expected)
-                    .ExpectOne(() => _cluster.Join(_selfAddress));
+                    .ExpectOneAsync(TimeSpan.FromMinutes(1), () => _cluster.Join(_selfAddress));
```
Comment (eaba) [inline src/core/Akka.Cluster.Tests/ClusterLogSpec.cs:68]: The default timeout was 3s....bumped this up to see how that goes!

**Comment by Aaronontheweb** [comment]
What's the error message this test usually fails with?

**Comment by eaba** [comment]
> What's the error message this test usually fails with?

```shell
Timeout (00:00:03) while waiting for messages. Only received 0/1 messages that matched filter [Info when Message contains " - event MemberUp"]

Expected: True

Actual:   False
```

```shell
at Akka.TestKit.Xunit2.XunitAssertions.Fail(String format, Object[] args) in D:\a\1\s\src\contrib\testkits\Akka.TestKit.Xunit2\XunitAssertions.cs:line 26

   at Akka.TestKit.Internal.InternalEventFilterApplier.Intercept[T](Func`1 func, ActorSystem system, Nullable`1 timeout, Nullable`1 expectedOccurrences, MatchedEventHandler matchedEventHandler) in D:\a\1\s\src\core\Akka.TestKit\EventFilter\Internal\EventFilterApplier.cs:line 309

   at Akka.TestKit.Internal.InternalEventFilterApplier.InternalExpect(Action action, ActorSystem actorSystem, Int32 expectedCount, Nullable`1 timeout) in D:\a\1\s\src\core\Akka.TestKit\EventFilter\Internal\EventFilterApplier.cs:line 451

   at Akka.TestKit.Internal.InternalEventFilterApplier.ExpectOne(Action action) in D:\a\1\s\src\core\Akka.TestKit\EventFilter\Internal\EventFilterApplier.cs:line 46

   at Akka.Cluster.Tests.ClusterLogSpec.<>c__DisplayClass9_0.<Join>b__0() in D:\a\1\s\src\core\Akka.Cluster.Tests\ClusterLogSpec.cs:line 65

   at Akka.TestKit.TestKitBase.<>c__DisplayClass158_0.<Within>b__0() in D:\a\1\s\src\core\Akka.TestKit\TestKitBase_Within.cs:line 57

   at Akka.TestKit.TestKitBase.Within[T](TimeSpan min, TimeSpan max, Func`1 function, String hint, Nullable`1 epsilonValue) in D:\a\1\s\src\core\Akka.TestKit\TestKitBase_Within.cs:line 134

   at Akka.TestKit.TestKitBase.Within(TimeSpan min, TimeSpan max, Action action, String hint, Nullable`1 epsilonValue) in D:\a\1\s\src\core\Akka.TestKit\TestKitBase_Within.cs:line 57

   at Akka.TestKit.TestKitBase.Within(TimeSpan max, Action action, Nullable`1 epsilonValue) in D:\a\1\s\src\core\Akka.TestKit\TestKitBase_Within.cs:line 32

   at Akka.Cluster.Tests.ClusterLogSpec.Join(String expected) in D:\a\1\s\src\core\Akka.Cluster.Tests\ClusterLogSpec.cs:line 63

   at Akka.Cluster.Tests.ClusterLogVerboseEnabledSpec.A_cluster_must_log_verbose_cluster_events_when_log_info_verbose_is_on() in D:\a\1\s\src\core\Akka.Cluster.Tests\ClusterLogSpec.cs:line 131
```

**Comment by Aaronontheweb** [comment]
LGTM

**Comment by Aaronontheweb** [comment]
This spec is still failing

**Pull request merged**